odoo/tools/xml_utils.py

```diff
@@ -340,3 +340,96 @@ def validate_xml_from_attachment(env, xml_content, xsd_name, reload_files_functi
 def find_xml_value(xpath, xml_element, namespaces=None):
     element = xml_element.xpath(xpath, namespaces=namespaces)
     return element[0].text if element else None
+
+
+def dict_to_xml(node, nsmap={}, template=None, render_empty_nodes=False, tag=None, path=None):
+    """ Helper to render a Python dict as an XML node.
+
+    The dict is expected to be of the form:
+    {
+        # Special keys:
+        '_tag': 'tag_name',  # '_tag' is rendered as the node's tag
+        '_text': 'content',  # '_text' is rendered as the node's text content
+        '_dummy': 'dummy_value',  # Keys starting with '_' are not rendered
+
+        # Simple values are rendered as attributes
+        'attribute_name': 'attribute_value',
+
+        # Dicts are rendered as child nodes
+        'child_tag': {
+            '_text': 'content',
+            'attribute_name': 'attribute_value',
+        },
+
+        # Lists of dicts are also rendered as child nodes
+        'child_tag': [
+            {
+                '_text': 'content',
+                'attribute_name': 'attribute_value',
+            },
+        ],
+    }
+
+    :param node: The Python dict to render.
+    :param nsmap: (optional) A dict of namespaces to be used for rendering the node.
+    :param template: (optional) A Python dict providing default values and an order of keys for rendering the node.
+    :param render_empty_nodes: (optional) If True, empty nodes will be rendered in the XML tree.
+    :param tag: (optional) The tag of the node to render (needed only for recursive calls).
+    :param path: (optional) The path of the currently rendered node in the XML tree (needed only for recursive calls).
+    :return: The rendered XML node as an lxml.Element.
+    """
+    def convert_tag_to_lxml_convention(tag):
+        if ':' in tag:
+            namespace, local_name = tag.split(':')
+            if namespace in nsmap:
+                return etree.QName(nsmap[namespace], local_name).text
+        return tag
+
+    if template is not None:
+        # Ensure order of keys
+        node = dict.fromkeys(template) | node
+
+    # Create the element
+    tag = node.get('_tag') or (template or {}).get('_tag', tag) or 'Node'
```

Comment:
Don't default to "Node", it is most probably an input error so you should raise ValueError.